auction-server/src/opportunity/repository/remove_opportunity.rs

```diff
@@ -17,8 +20,9 @@ impl<T: InMemoryStore> Repository<T> {
         &self,
         db: &sqlx::Pool<Postgres>,
         opportunity: &T::Opportunity,
-        reason: OpportunityRemovalReason,
+        reason: entities::OpportunityRemovalReason,
```

Comment:
maybe `reason: impl Into<OpportunityRemovalReason>,` ? if you just want it to be convertable

Reply:
I want to get the removal reason type of the entities :-?
We should use entities or primary types in repo functions